apps/user-office-frontend/src/components/questionary/questionaryComponents/ProposalBasis/QuestionaryComponentProposalBasis.tsx

```diff
@@ -45,18 +46,21 @@ function QuestionaryComponentProposalBasis(props: BasicComponentProps) {
 
   const [localTitle, setLocalTitle] = useState(state?.proposal.title);
   const [localAbstract, setLocalAbstract] = useState(state?.proposal.abstract);
+  const { user } = useContext(UserContext);
 
   if (!state || !dispatch) {
     throw new Error(createMissingContextErrorMessage());
   }
 
   const { proposer, users } = state.proposal;
+  const { loading, userData } = useBasicUserData(user.id); // Get basicUserData from the logged in user
+  const [piData, setPIData] = useState<BasicUserDetails | null>(null);
 
-  const [principalInvestigatorUserId, setPrincipalInvestigatorUserId] =
-    useState<number | undefined>(
-      state?.proposal.proposer?.id //user that is logged in
-    );
-  const { loading, userData } = useBasicUserData(principalInvestigatorUserId);
+  useEffect(() => {
+    if (userData !== null) {
+      setPIData(userData);

```

Comment:
There is a slight problem with it, because if you edit a proposal as User Officer it will set the PI as tie user officer

Reply:
Works :+1: